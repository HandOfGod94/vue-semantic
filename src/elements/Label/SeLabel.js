import Vue from 'vue'
import Component from 'vue-class-component'

@Component({
  name: 'se-label',
  props: {
    as: { type: String, default: 'div' },
    image: Boolean,
    color: String,
    basic: Boolean,
    empty: Boolean,
    circular: Boolean,
    size: String,
    corner: String,
    tag: Boolean,
    ribbon: Boolean,
    pointing: String,
    horizontal: Boolean,
    floating: Boolean,
    attach: String
  }
})
class SeLabel extends Vue {
  render (h) {
    let Label = this.as
    return (
      <Label class={this.classes}>
        {this.$slots.default}
      </Label>
    )
  }

  get classes () {
    let classes = {
      floating: this.floating,

      ui: true,

      [this.size]: this.size,
      image: this.image,
      tag: this.tag,
      ribbon: this.ribbon,
      [this.color]: this.color,
      empty: this.empty,
      horizontal: this.horizontal,
      circular: this.circular,
      basic: this.basic,
      [`${this.corner} corner`]: this.corner,
      [`${this.pointing} pointing`]: this.pointing,
      [`${this.attach} attached`]: this.attach,

      label: true
    }

    return classes
  }
}

export default SeLabel
