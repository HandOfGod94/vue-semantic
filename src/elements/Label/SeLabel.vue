<template>
  <component :is="as" :class="classObject">
    <slot></slot>
    <div v-if="hasDetail" class="detail">
      <slot name="detail">{{detail}}</slot>
    </div>
  </component>
</template>

<script>
export default {
  name: 'se-label',
  props: {
    as: { type: String, default: 'div' },
    image: Boolean,
    color: String,
    basic: Boolean,
    empty: Boolean,
    circular: Boolean,
    size: String,
    position: String,
    tag: Boolean,
    ribbon: Boolean,
    pointing: Boolean,
    horizontal: Boolean,
    floating: Boolean,
    attach: Boolean,
    detail: String
  },
  computed: {
    classObject () {
      return {
        floating: this.floating,
        ui: true,
        [this.size]: this.size,
        image: this.image,
        tag: this.tag,
        ribbon: this.ribbon,
        [this.color]: this.color,
        empty: this.empty,
        horizontal: this.horizontal,
        circular: this.circular,
        basic: this.basic,
        [this.position]: this.position,
        pointing: this.pointing,
        attached: this.attach,
        label: true
      }
    },
    hasDetail () { return !!this.$slots['detail'] || this.detail }
  }
}
</script>

<style>

</style>
