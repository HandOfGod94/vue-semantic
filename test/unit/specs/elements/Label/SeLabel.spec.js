import SeLabel from '@/elements/Label/SeLabel'
import { mount } from '@vue/test-utils'

describe('SeLabel snapshot test', () => {
  it('should match with existing snapshot', () => {
    const wrapper = mount(SeLabel)
    expect(wrapper.classes()).toContain('ui')
    expect(wrapper.classes()).toContain('label')
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('SeLabel classes test', () => {
  it('should have "floating" class', () => {
    const data = { propsData: { floating: true } }
    const wrapper = mount(SeLabel, data)
    expect(wrapper.classes()).toContain('floating')
  })

  it('should have "tag" class', () => {
    const data = { propsData: { tag: true } }
    const wrapper = mount(SeLabel, data)
    expect(wrapper.classes()).toContain('tag')
  })

  it('should have "ribbon" class', () => {
    const data = { propsData: { ribbon: true } }
    const wrapper = mount(SeLabel, data)
    expect(wrapper.classes()).toContain('ribbon')
  })

  it('should have "empty" class', () => {
    const data = { propsData: { empty: true } }
    const wrapper = mount(SeLabel, data)
    expect(wrapper.classes()).toContain('empty')
  })

  it('should have "corner" class', () => {
    const data = { propsData: { corner: 'left' } }
    const wrapper = mount(SeLabel, data)
    expect(wrapper.classes()).toContain('left')
    expect(wrapper.classes()).toContain('corner')
  })
})

describe('SeLabel mixin classes test', () => {
  it('should have "basic" class', () => {
    const data = { propsData: { basic: true } }
    const wrapper = mount(SeLabel, data)
    expect(wrapper.classes()).toContain('basic')
  })

  it('should have "circular" class', () => {
    const data = { propsData: { circular: true } }
    const wrapper = mount(SeLabel, data)
    expect(wrapper.classes()).toContain('circular')
  })

  it('should have "red" class for color', () => {
    const data = { propsData: { color: 'red' } }
    const wrapper = mount(SeLabel, data)
    expect(wrapper.classes()).toContain('red')
  })

  it('should have "mini" class for size', () => {
    const data = { propsData: { size: 'mini' } }
    const wrapper = mount(SeLabel, data)
    expect(wrapper.classes()).toContain('mini')
  })
})
