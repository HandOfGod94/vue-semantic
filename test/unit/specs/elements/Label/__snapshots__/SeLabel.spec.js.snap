// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeLabel snapshot test should match with existing snapshot 1`] = `
<div class="ui label">
    <!---->
</div>
`;
